### PR TITLE
Cherrypick 1.2.x

### PR DESCRIPTION
**Pull request by SteveDiamond**
## Description
Cherry picks to prepare for new patch

#1931 
#1951 
#1866 
#1962 
Elements of #1978 
#1984
#1987

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [x] Bug fix
- [ ] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [ ] Add our license to new files.
- [ ] Check that your code adheres to our coding style.
- [ ] Write unittests.
- [ ] Run the unittests and check that they’re passing.
- [ ] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

**Comment by phschiele** [comment]
Double-checked that this contains all bug fixes since the last cherry-pick PR.

File: cvxpy/reductions/complex2real/complex2real.py [hunk 1]
```diff
@@ -34,6 +35,9 @@ def accepts(problem) -> bool:
 class Complex2Real(Reduction):
     """Lifts complex numbers to a real representation."""
 
+    UNIMPLEMENTED_REAL_DUALS = (PowCone3D, PowConeND)
```
Comment (rileyjmurray) [inline cvxpy/reductions/complex2real/complex2real.py:38]: I believe PowCone3D does have real duals implemented.

File: cvxpy/reductions/complex2real/complex2real.py [hunk 2]
```diff
@@ -34,6 +35,9 @@ def accepts(problem) -> bool:
...
+    UNIMPLEMENTED_COMPLEX_DUALS = (SOC, ExpCone)
```
Comment (rileyjmurray) [inline cvxpy/reductions/complex2real/complex2real.py:39]: ExpCone constraints can never be complex, so no need to list here.

File: cvxpy/reductions/complex2real/complex2real.py (hunk 1 above)
Comment (rileyjmurray) [reply cvxpy/reductions/complex2real/complex2real.py:38]: I'm getting an error with PowCone3D, suggesting that this chance is needed, but I don't know why it's needed. Investigating now.

**Pull request merged**